File: src/main/java/co/aplicared/jvm/juego/arbúcies/level/TileCoord.java
package co.aplicared.jvm.juego.arbúcies.level;

public class TileCoord {
    private final int TILE_SIZE = 16;
    private int x, y;

    public TileCoord(int x, int y) {
        this.x = x * TILE_SIZE;
        this.y = y * TILE_SIZE;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x * TILE_SIZE;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y * TILE_SIZE;
    }

    public int[] getArray() {
        int[] r = new int[2];
        r[0] = x;
        r[1] = y;
        return r;
    }
}

File: src/main/kotlin/co/aplicared/jvm/juego/arbucies/level/TileCoord.kt
package co.aplicared.jvm.juego.arbucies.level

open class TileCoord(x: Int, y: Int) {
    private val _tileSize = 16

    var x: Int = x * _tileSize
        set(value) {
            field = value * _tileSize
        }

    var y: Int = y * _tileSize
        set(value) {
            field = value * _tileSize
        }

    fun getArray(): IntArray {
        val r = IntArray(2)
        r[0] = x
        r[1] = y
        return r
    }
}